graph/push.go

```diff
@@ -25,6 +31,7 @@ type Pusher interface {
 	Push() (fallback bool, err error)
 }
 
+// NewPusher return a new implementation conforming to Pusher interface.
```

Comment:
s/return/returns/

You should also add the it checks for the registry version.